src/zenml/client.py

```diff
@@ -4438,23 +4438,20 @@ def _delete_artifact_from_artifact_store(
     def create_run_metadata(
         self,
         metadata: Dict[str, "MetadataType"],
-        resource_id: UUID,
-        resource_type: MetadataResourceTypes,
+        resources: List[RunMetadataResource],
         stack_component_id: Optional[UUID] = None,
+        cached: bool = False,
     ) -> None:
         """Create run metadata.
 
         Args:
             metadata: The metadata to create as a dictionary of key-value pairs.
-            resource_id: The ID of the resource for which the
-                metadata was produced.
-            resource_type: The type of the resource for which the
+            resources: The list of IDs and types of the resources for that the
                 metadata was produced.
             stack_component_id: The ID of the stack component that produced
                 the metadata.
-
-        Returns:
-            None
+            cached: A flag indicating if the run metadata can be cached during
```

Comment:
It doesn't feel right to have this as a public option in the client method somehow. If anything, this should be a server-internal flag that is somehow set from the context. I think we talked about it and it is used to differentiate metadata that was logged to a step after its execution, which should not be part of the metadata if we create a cached step from it, correct?

Reply:
That's correct regarding the use-case. 

I also don't like the fact that it is a part of our public client, but unfortunately, we can not handle it from the server side at the moment. I will keep this on the backburner but for now I'm resolving this comment.